src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -1011,6 +1023,40 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
   // END extra overrides
   ///////////////////////////////////////////////////////////////////
 
+  override private[collection] def filterInPlaceImpl(p: (K, V) => Boolean): this.type = {
+    val it = readOnlySnapshot().iterator
+    while (it.hasNext) {
+      val (k, v) = it.next()
+      if (p(k, v)) removeRefEq(k, v)
```

Comment:
CONDITION IS INVERTED!!!!!!!!!

```suggestion
      if (!p(k, v)) removeRefEq(k, v)
```